x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/GrokExec.java

```diff
@@ -31,6 +42,31 @@ public GrokExec(
         this.parser = parser;
     }
 
+    private static GrokExec readFrom(StreamInput in) throws IOException {
```

Comment:
I'm curious why you used a private static factory function here instead of a constructor like in all other classes?

Reply:
It needs `source` in two places and those places aren't really compatible with a ctor. It's funky, but I figure I'll use whatever takes the minimum amount of typing.